rs/ic_os/config/src/update_config.rs

```diff
@@ -0,0 +1,574 @@
+use crate::serialize_and_write_config;
+use anyhow::{Context, Result};
+use mac_address::mac_address::FormattedMacAddress;
+use serde_json;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs;
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::path::Path;
+use url::Url;
+
+use crate::types::*;
+
+pub static CONFIG_ROOT: &str = "/boot/config";
+pub static STATE_ROOT: &str = "/var/lib/ic/data";
+
+pub fn update_guestos_config() -> Result<()> {
+    let config_dir = Path::new(CONFIG_ROOT);
+    log_directory_structure(config_dir)?;
+    let state_root = Path::new(STATE_ROOT);
+    log_directory_structure(state_root)?;
```

Comment:
This has a chance to get really really large. What do we need to display out of this dir? If everything is top level, we should limit the depth.

Reply:
Changed to log just the top level 👍